test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightBatchReaderTests.cs

```diff
@@ -363,6 +363,109 @@ public async Task ReadBatchRequestWithChangesetAndDependsOnIdsAsync()
                 isResponse: false);
         }
 
+        [Fact]
+        public async Task ReadBatchRequestWithDuplicateAttributesAsync()
+        {
+            var payload = "{\"requests\": [{" +
+                "\"id\": \"1\"," +
+                "\"method\": \"GET\"," +
+                "\"method\": \"POST\"," +
+                "\"url\": \"http://tempuri.org/Customers\"," +
+                "\"headers\": {\"odata-version\":\"4.0\",\"content-type\":\"application/json;odata.metadata=minimal;odata.streaming=true;IEEE754Compatible=false;charset=utf-8\"}, " +
+                "\"body\": {\"@odata.type\":\"#NS.Customer\",\"Id\":1,\"Name\":\"Customer 1\",\"Type\":\"Retail\"}}]}";
+
+            await SetupJsonLightBatchReaderAndRunTestAsync(
+                payload,
+                async (jsonLightBatchReader) =>
+                {
+                    try
+                    {
+                        while (await jsonLightBatchReader.ReadAsync())
+                        {
+                            if (jsonLightBatchReader.State == ODataBatchReaderState.Operation)
+                            {
+                                await jsonLightBatchReader.CreateOperationRequestMessageAsync();
+                            }
+                        }
+
+                        Assert.True(true, "The test was successful, because no ArgumentException was thrown");
+                    }
+                    catch (ArgumentException ex)
+                    {
+                        Assert.False(true, ex.Message);
+                    }
+                },
+                isResponse: false);
+        }
+
+        [Fact]
+        public async Task ReadBatchRequestWithDuplicateHeadersAsync()
+        {
+            var payload = "{\"requests\": [{" +
+                "\"id\": \"1\"," +
+                "\"method\": \"POST\"," +
+                "\"url\": \"http://tempuri.org/Customers\"," +
+                "\"headers\": {\"odata-version\":\"4.0\",\"content-type\":\"application/json;odata.metadata=minimal;odata.streaming=true;IEEE754Compatible=false;charset=utf-8\",\"duplicate-header\":\"value1\",\"duplicate-header\":\"value2\"}, " +

```

Comment:
@habbes I am sorry, but we have no impact on the incoming requests and as shown in the test data, the requests show duplicate headers, which currently result in HTTP 500 responses.
If decided that duplicate headers are not allowed, we'd like to see this restriction being documented in the OData Specifications [Batch Request Headers](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_BatchRequestHeaders) and handled by a specific HTTP 4XX response.
But as a first approach, I tried to implement the solution, as shown in [Issue-2656](https://github.com/OData/odata.net/issues/2656) for handling duplicate cookie headers.

Reply:
The current code implements a "last entry wins" semantic. This is different than what I was suggesting which was, *if* we wanted to support duplicate headers, that we align with the semantics for multiple occurrences of the same headers defined in RFC2616 for an HTTP request which specifies that the same header can appear only if the value is a comma separated list, in which case the values are appended.

Which is not exactly the same as having multiple JSON entries with the same value (which is not supported for interoperable JSON clients), but was the closest semantics I could come up with for handling duplicate headers.

I brought this up in our OASIS OData Technical Meeting today (see [https://issues.oasis-open.org/browse/ODATA-1581?jql=project%20%3D%20ODATA](https://issues.oasis-open.org/browse/ODATA-1581?jql=project%20%3D%20ODATA)) and we agreed to clarify in the spec that clients should not have duplicate properties in the batch request, nor duplicate headers in the header collection, and that services *should* return a client error for these cases. 

This is a SHOULD, not a MUST, since some services may already have other behavior, but now that the direction has been defined in the standard we should align and return a client error for duplicate headers (as we do for duplicate request properties).

Reply:
Understood, I'll prepare the necessary changes.
Are explicit test cases necessary, that proof duplicate attributes & headers throw the client error?